### PR TITLE
Fix post survey urls that rely on day

### PR DESCRIPTION
**Pull request by molly-moen**
Some of our post survey logic pulls the session count and uses that to determine the day to send in a url/as a parameter. This can cause issues if a workshop is of an irregular length. The simple fix here is to fall back to the largest valid day we have for a workshop if the actual workshop is too long. 
Found this issue due to a 1 day workshop that was split into 2 days, which caused the honeybadger errors linked below.

## Links
- [honeybadger 1](https://app.honeybadger.io/projects/3240/faults/55995670)
- [honeybadger 2](https://app.honeybadger.io/projects/3240/faults/62994757)

## Testing story
Made a workshop with the same setup as the one that caused the honeybadger errors locally and verified the fix causes the links and survey submissions to work where the didn't before. Also added a unit test for the enrollments model to cover this case.

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/app/controllers/pd/workshop_daily_survey_controller.rb [hunk 1]
```diff
@@ -496,6 +496,14 @@ def get_session_for_workshop_and_day(workshop, day)
       day > 0 ? workshop.sessions[day - 1] : nil
     end
 
+    def get_last_valid_day(workshop)
```
Comment (bencodeorg) [inline dashboard/app/controllers/pd/workshop_daily_survey_controller.rb]: Could this just be a method on the workshop model and you reference it like `workshop.last_valid_day`?

Comment (breville) [reply dashboard/app/controllers/pd/workshop_daily_survey_controller.rb]: +1 for factoring this common code, and putting it in the model.  (When in doubt, put business logic in the model.)

Comment (molly-moen) [reply dashboard/app/controllers/pd/workshop_daily_survey_controller.rb]: good point

**Review by breville — APPROVED** [review]

**Review by bencodeorg — APPROVED** [review]
👍

**Pull request merged**